src/common/snap_types.cc

```diff
@@ -53,6 +53,62 @@ void SnapRealmInfo::generate_test_instances(std::list<SnapRealmInfo*>& o)
   o.back()->prior_parent_snaps.push_back(5);
 }
 
+// -- "new" SnapRealmInfo --
+
+void SnapRealmInfoNew::encode(ceph::buffer::list& bl) const
+{
+  using ceph::encode;
+  ENCODE_START(1, 1, bl);
+  encode(info, bl);
+  encode(last_modified, bl);
+  encode(change_attr, bl);
+  ENCODE_FINISH(bl);
+}
+
+void SnapRealmInfoNew::decode(ceph::buffer::list::const_iterator& bl)
+{
+  using ceph::decode;
+  DECODE_START(1, bl);
+  decode(info, bl);
+  decode(last_modified, bl);
+  decode(change_attr, bl);
+  DECODE_FINISH(bl);
+}
+
+void SnapRealmInfoNew::dump(ceph::Formatter *f) const
+{
+  f->dump_unsigned("ino", ino());
+  f->dump_unsigned("parent", parent());
+  f->dump_unsigned("seq", seq());
+  f->dump_unsigned("parent_since", parent_since());
+  f->dump_unsigned("created", created());
+  f->dump_stream("last_modified") << last_modified;
+  f->dump_unsigned("change_attr", change_attr);
+
+  f->open_array_section("snaps");
+  for (auto p : info.my_snaps) {
+    f->dump_unsigned("snap", p);
+  }
+  f->close_section();
+
+  f->open_array_section("prior_parent_snaps");
+  for (auto p : info.prior_parent_snaps) {
+    f->dump_unsigned("snap", p);
+  }
+  f->close_section();
+}
```

Comment:
Could this be just:

```
void SnapRealmInfoNew::dump(ceph::Formatter *f) const
{
  info.dump(f);
  f->dump_stream("last_modified") << last_modified;
  f->dump_unsigned("change_attr", change_attr);
}

```

Reply:
I missed doing that - thx for catching!

Reply:
I recall why I didn't do it this way - I wanted the new fields to show up dumping snaps and prior parent snaps. I'll make the  change since its code duplication.